### PR TITLE
Update imports to work with later version of PyTorch Lightning

### PR DESCRIPTION
**Pull request by nathancooperjones**
## Describe the big picture - why are we doing this?
Currently, versions of PyTorch Lightning equal or later than `1.6.0` are not compatible with Collie due to them changing their file structure for a couple imports. For these versions, the error would be cryptic and would avoid importing any function from Collie at all. The only fix would be installing to an earlier version of PyTorch Lightning, which is not what we want. 

The solution to this is to just add in additional `try/except ImportError` statements to relevant files so new _and_ older versions of PyTorch Lightning work with Collie. There are also a couple of spots in the codebase where some data has to be moved to the model device in a slightly updated way to work with newer versions of PyTorch Lightning, which is also a simple addition. 

**All changes made in this PR are NOT breaking changes and should be compatible with both past and current versions of PyTorch Lightning. Merging this is will NOT break existing Collie workflows** 

## Any additional details to clarify code in the PR?
In addition, I went ahead and updated the requirements files. I tested that this works with the current Docker setup and confirmed that tests still pass. 

## How is this tested?
Tested with unit tests and building Docker containers with PyTorch Lightning versions `1.5.7` and `1.9.4` on both CPU and CUDA PyTorch environments. 

## Pull Request Checklist
 - [X] Pull request includes a description of why we are doing this
 - [x] ``CHANGELOG`` has been updated
 - [x] Version in ``_version.py`` has been updated
 - [X] All tests in the ``tests`` folder pass with a local build
 - [ ] ``README`` has been updated (if applicable)
 - [ ] Documentation in ``docs`` has been updated (if applicable)
 - [X] ``requirements.txt`` and ``requirements-dev.txt`` have been recompiled (if applicable)
 - [X] Docker image can be built using ``docker build -t collie .``

### CODE REVIEW

File: .codecov.yaml [hunk 1]
```diff
@@ -4,8 +4,3 @@ coverage:
       default:
         target: 90%  # we want at least 90% coverage in Collie
         if_ci_failed: error
-    patch:
-      default:
-        target: auto
-        threshold: 5%  # PR may only decrease test coverage by up to 5%
-        if_ci_failed: error
```
Comment (nathancooperjones) [inline .codecov.yaml:11]: This PR adds so few lines of code, but with most of them being `try/else` for PyTorch Lightning version things, the patch difference made the CI checks fail. I'm okay removing this and just keeping the overall target 90% test coverage instead.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -10,7 +10,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ['3.6', '3.7', '3.8', '3.9']
```
Comment (nathancooperjones) [inline .github/workflows/ci.yaml:13]: Python 3.6 has reached end-of-life - we no longer need to test for it.

File: tests/fixtures/model_fixtures.py [hunk 3]
```diff
@@ -494,19 +494,15 @@ def models_trained_for_one_step(request,
                                   logger=False,
                                   enable_checkpointing=False)
 
-    if request.param == 'mf_no_val':
-        with pytest.warns(UserWarning):
```
Comment (nathancooperjones) [inline tests/fixtures/model_fixtures.py:498]: With newer versions of PyTorch Lightning, it's no longer a guarantee this warning is raised more than once. For that reason, we just remove the check for a warning to be raised and proceed with fitting as normal.

**Comment by nathancooperjones** [comment]
This PR won't pass the `codecov/patch` CI check, but that is expected given the weird nature of this PR.

**Review by slivelsbergersr — APPROVED** [review]

File: collie/model/base/base_pipeline.py [hunk 4]
```diff
@@ -268,7 +268,11 @@ def forward(self, users: torch.tensor, items: torch.tensor) -> torch.tensor:
 
     def _move_any_external_data_to_device(self):
         """Code for ensuring all side-data is put onto the model's device before training."""
-        pass
+        if self.hparams.metadata_for_loss is not None:
```
Comment (slivelsbergersr) [inline collie/model/base/base_pipeline.py:271]: Read our minds. We were just realizing we wanted to make this update the other day.

**Pull request merged**